### PR TITLE
Remove CommParams and make the fields there data attributes.

### PR DESCRIPTION
**Pull request by wujingyue**
I initially intended to just make CommParams a data attribute. Then I realized the extra indirection of CommParams doesn't seem to give much practical value after various cleanups we've done, so I flattened CommParams as well and that saved a bit of typing.

### CODE REVIEW

File: csrc/multidevice/communication.h [hunk 1]
```diff
@@ -34,36 +34,87 @@ enum class CommunicationType {
...
+  int64_t getRootRelativeIndex();
+
+ private:
+  // This can be computed from params_, but given how frequently this is used
+  // in the hot path, I'm currently storing it as a field that'll be computed by
+  // Communication::team().
+  Team team_;
+};
 
 // The method "post" triggers the execution of the communication. This call is
```
Comment (wujingyue) [inline csrc/multidevice/communication.h:112]: Note: I moved this comment to postSingleCommunication, because Communication::post is no longer a thing.

**Comment by wujingyue** [comment]
!build --dist

**Comment by xwang233** [comment]
!build --dist

**Review by samnordmann — APPROVED** [review]
LGTM! Thanks, this makes Communication IRs cleaner

**Comment by wujingyue** [comment]
!build

**Pull request merged**